app/models/cloud_network.rb

```diff
@@ -50,6 +50,10 @@ def self.class_by_ems(ext_management_system, _external = false)
     ext_management_system && ext_management_system.class::CloudNetwork
   end
 
+  def self.tenant_id_clause_format(tenant_ids)
+    ["((tenants.id IN (?) OR cloud_networks.shared IS TRUE) AND ext_management_systems.tenant_mapping_enabled IS TRUE) OR ext_management_systems.tenant_mapping_enabled IS FALSE OR ext_management_systems.tenant_mapping_enabled IS NULL", tenant_ids]
```

Comment:
Where does `tenants` come from?  I see a `belongs_to :cloud_network`

Reply:
@bdunne The reference comes from the included CloudTenancyMixin -- this method overrides tenant_id_clause_format in the mixin to add the part about cloud_networks, but the rest is right from there. As for the actual source of the tenants table in the query mix, it looks like that comes from this reference in cloud_tenant.rb: "has_one    :source_tenant, :as => :source, :class_name => 'Tenant' "

Reply:
@bdunne Any other issues with the PR?

Reply:
hi @bdunne, anything @sseago needs to change about this PR?

Reply:
Can we write this in AREL? We try to avoid raw sql when possible.

Reply:
I've got nothing else.

Reply:
@Ladas I was trying to keep the implementation as close as possible to the method I'm overriding, which currently uses raw sql. If that gets updated to use AREL, then this should be modified to do the same: https://github.com/ManageIQ/manageiq/blob/master/app/models/mixins/cloud_tenancy_mixin.rb#L16-L18

Reply:
Sounds good then, the refactoring should start in the original method. :+1:

Reply:
In light of @Ladas's comment, can this be merged?

Reply:
ping @Ladas @bdunne ^ https://github.com/ManageIQ/manageiq/pull/17305#discussion_r189612598

Reply:
I don't have merge rights for this repo, @bdunne ?